hl7/4.2/modules/ROOT/edits/index.adoc

```diff
@@ -0,0 +1,174 @@
+structure, your schema location is `/hl7/v2_5_1/ADT_A01.esl` for the
+standard version or `/hl7lax/v2_5_1/ADT_A01.esl` for the lax version.
+
+If you're using one or more custom schemas, put these under
+a directory in `src/main/mule` and refer to the location using `${app.home}`.
+For example, if you've put your ADT_A01 schema under `src/main/mule/mypartner/ADT_A01.esl,
+your schema location is `${app.home}/mypartner/ADT_A01.esl`.
+
+The Mule runtime engine automatically checks `src/main/mule` for any locations
+that contain the `${app.home}` value.
+
+== The Hierarchy of X12 EDI Connector Message Structure
+
+Use this connector to read or write HL7 documents into (or from) the canonical ER7 message structure.
+This structure is a hierarchy of Java Maps and Lists. You use either DataWeave or code to manipulate them.
+Each transaction structure is defined in the schemas.
+
+The message itself contains the following keys, some of which apply only to either the read operation or the write operation, as indicated:
+
+[%header%autowidth.spread]
+|===
+|Key name |Description
+|ACK (read only) |Message generated by the module during the read operation. The `MSA-01` acknowledgment code value is based on the parser configuration settings.
+|Data (read or write) |Wrapper for message data, with a key matching the message structure ID value that links to the actual data. Allows different messages to be included in the metadata and handled in DataWeave mappings.
+|Delimiters (read or write) |The delimiters used for the message.
+The characters in the string of delimiters are interpreted based on their position in string, in the following order: component separator, repetition separator, escape character, and subcomponent separator, with a value of `U` indicating no value in that position.
+|Errors (read only) |A list of errors associated with the input message.
+|Id |Message structure ID
+|MSH (read only) |Link that receives MSH segment data
+|Name (read only) |Message structure name
+|===
+
+Individual messages have their own maps, with keys matching the segments of the message. For example, an ACK message uses the message structure ID ACK, and the data for the sent or received ACK message is an ACK value in the Data map. The ACK message is itself a map, and the segments and groups of the message are represented as maps (in the case of singleton instances) or lists of maps (for repeating instances) with positional keys.
+
+There are two special cases where using generic handling for data that is not included in a schema definition.
+
+The first is for HL7 values of the `varies` type. Since these values consist of any structure of components and subcomponents that can be repeated, the parser uses a list of map representations for each `varies` of type. The keys in each map are generated as the value is parsed, matching standard HL7 value names with two digits used for each nesting level.
```

Comment:
```suggestion
The first case is HL7 values of the `varies` type. Because these values consist of any structure of components and subcomponents that can be repeated, the parser uses a list of map representations for each `varies` type. The keys in each map are generated as the value is parsed, matching standard HL7 value names with two digits used for each nesting level.
```